### PR TITLE
Refactor to remove querySelector calls where possible

### PR DESCRIPTION
**Pull request by codyarose**

### CODE REVIEW

File: src/__tests__/medications/ViewMedication.test.tsx [hunk 1]
```diff
@@ -116,11 +116,11 @@ describe('View Medication', () => {
...
   it('should not display the canceled date if the medication is not canceled', async () => {
-    const { container } = setup({} as Medication, [Permissions.ViewMedication])
+    setup({} as Medication, [Permissions.ViewMedication])
 
-    await waitFor(() => {
-      expect(container.querySelector('.canceled-on')).not.toBeInTheDocument()
-    })
+    expect(
+      screen.queryByRole('heading', { name: /medications\.medication\.canceledOn/i }),
+    ).not.toBeInTheDocument()
```
Comment (nobrayner) [inline src/__tests__/medications/ViewMedication.test.tsx:123]: I'm a little worried there may be a spinner or something here making this test not actually doing what you may think.

If there isn't a spinner, then ignore me :P

Comment (codyarose) [reply src/__tests__/medications/ViewMedication.test.tsx:123]: Nope, doesn't appear to be a spinner in this one!

File: src/__tests__/patients/related-persons/RelatedPersonsTab.test.tsx [hunk 2]
```diff
@@ -77,10 +77,9 @@ describe('Related Persons Tab', () => {
     })
 
     it('should not render a New Related Person button if the user does not have write privileges for a patient', async () => {
-      const { container } = setup({ permissions: [Permissions.ReadPatients] })
-
-      await waitForElementToBeRemoved(container.querySelector(`[class^='css']`))
+      setup({ permissions: [Permissions.ReadPatients] })
 
+      expect(await screen.findByRole('alert')).toBeInTheDocument()
```
Comment (nobrayner) [inline src/__tests__/patients/related-persons/RelatedPersonsTab.test.tsx]: Probably getting nit-picky on this one. Similar to https://github.com/kcdraidgroup/hospitalrun-frontend/pull/273/files#r557056312 - the semantics of this no longer say "wait for the css spinner to disappear before checking the button isn't there".

As it stands with the alert, I am now led to believe the alert is somehow important to this test.

File: src/__tests__/patients/care-goals/CareGoalTab.test.tsx [hunk 3]
```diff
@@ -63,10 +63,9 @@ const setup = (
 
 describe('Care Goals Tab', () => {
   it('should not render add care goal button if user does not have permissions', async () => {
-    const { container } = setup('/patients/123/care-goals', [])
-
-    await waitForElementToBeRemoved(container.querySelector('.css-0'))
+    setup('/patients/123/care-goals', [])
 
+    expect(await screen.findByRole('table')).toBeInTheDocument()
     expect(screen.queryByRole('button', { name: /patient.careGoal.new/i })).not.toBeInTheDocument()
```
Comment (nobrayner) [inline src/__tests__/patients/care-goals/CareGoalTab.test.tsx]: https://github.com/kcdraidgroup/hospitalrun-frontend/pull/273/files#r557058803

**Review by nobrayner — APPROVED** [review]

**Pull request merged**